### PR TITLE
feat: remove app/main from URL structure

### PR DESCRIPTION
**Pull request by fiam**
Fixes ENG-701, ENG-703, ENG-704, ENG-705, ENG-706, ENG-730

- feat: remove /<app-name>/main prefix from router and pb definition
- chore(docs): update docs to reflect changes to WunderGraph protocol
- chore(docs): update docs to reflect changes to WunderGraph protocol
- fix: update code generation (#333)
- feat: update URLs in testapps for new style
- feat: remove /<app-name>/main prefix from router and pb definition
- fix: update code generation (#333)
- feat: drop Client applicationPath and associated properties
- fix: port apollo-federation example to new URL/API
- fix: update /auth0-oidc-authentication to new URL/API
- fix: update caching example to new URL/API
- fix: update cross-api-joins to new URL/API
- fix: update faunadb-nextjs to new URL/API
- fix: update fragments example to new URL/API
- fix: update golang-client example to new URL/API
- fix: update graphql-sse-subscriptions example to new URL/API
- fix: update graphql-subscriptions-hooks example to new URL/API
- fix: update graphql-ws-subscriptions example to new URL/API
- fix: update graphql-yoga-sse-subscriptions example to new URL/API
- fix: update hooks example to new URL/API
- fix: update inject-bearer example to new URL/API
- fix: update keycloak-oidc-authentication example to new URL/API
- fix: update migrate-from-apollo example to new URL/API
- fix: update nextjs example to new URL/API
- fix: update nextjs-postgres-prisma to new URL/API
- fix: update nextjs-swr example to new URL/API
- fix: update rbac example to new URL/API
- fix: upadte  schema-extension example to use new URL/API
- fix: update simple example to new URL/API
- fix: update webhooks example to new URL/API
- fix: update URLs in README files for new style
- fix: update publish-install-api to new URL/API
- fix: update graphql-hasura-subscriptions to new URL/API
- fix: permissions in graphql-apollo-subscriptions example
- fix: update graphql-apollo-subscriptions example to new URL/API
- fix: typo in graphql-apollo-subscriptions README
- fix: update postgres example to new URL/API
- fix: update nextjs-postgres-user-filters example to new URL/API
- docs: fix old-style URLs in a couple of pages
- feat: add redirect for old-style URLs

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]
LGTM

File: pkg/apihandler/apihandler.go [hunk 1]
```diff
@@ -187,10 +186,20 @@ func (r *Builder) BuildAndMountApiHandler(ctx context.Context, router *mux.Route
 	// limiter := rate.NewLimiter(rate.Every(time.Second), 10)
 
 	r.log.Debug("configuring API",
-		zap.String("name", api.PathPrefix),
 		zap.Int("numOfOperations", len(api.Operations)),
 	)
 
+	// Redirect from old-style URL, for temporary backwards compatibility
+	r.router.MatcherFunc(func(r *http.Request, rm *mux.RouteMatch) bool {
```
Comment (StarpTech) [inline pkg/apihandler/apihandler.go:193]: I thought, we won't do it?

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Comment by StarpTech** [comment]
Did you also consider the wg templates in wunderctl?

File: pkg/apihandler/apihandler.go (hunk 1 above)
Comment (fiam) [reply pkg/apihandler/apihandler.go:193]: We decided to add it for a few releases, to help users that might not read the release notes. If the old style URL is used, a warning is printed in the logs to give them a heads up.

**Comment by fiam** [comment]
> Did you also consider the wg templates in wunderctl?

No, I didn't update those. I understand that those are going away as soon as the new create-wundergraph-app is merged.

File: pkg/apihandler/apihandler.go (hunk 1 above)
Comment (StarpTech) [reply pkg/apihandler/apihandler.go:193]: ok

**Comment by StarpTech** [comment]
> No, I didn't update those. I understand that those are going away as soon as the new create-wundergraph-app is merged.

Yes, but then you need to coordinate with @JivusAyrus and only merge it after that.

**Comment by fiam** [comment]
> > No, I didn't update those. I understand that those are going away as soon as the new create-wundergraph-app is merged.
> 
> 
> 
> Yes, but then you need to coordinate with @JivusAyrus and only merge it after that.

No problem, I will wait for Suvij's PR to be merged.

**Review by StarpTech — APPROVED** [review]
LGTM, but please coordinate with @JivusAyrus before merge

**Pull request merged**